packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1639,7 +1646,24 @@ function updateSuspenseComponent(
       primaryChildFragment.sibling = fallbackChildFragment;
       // Skip the primary children, and continue working on the
       // fallback children.
-      workInProgress.memoizedState = SUSPENDED_MARKER;
+
+      if (workInProgress.effectTag & SoftCapture) {
+        // don't clear the effectTag yet, it'll be read in completeWork
+        workInProgress.effectTag &= ~SoftCapture;
+        const avoidedRenderExpiration = renderExpirationTime - 1; // verify/rename
```

Comment:
What if the renderExpirationTime is Never? Then this will yield NoWork. It overflows. What prevents this from happening? If something is preventing it, then we should probably assert that. E.g. a DEV error.